### PR TITLE
Make Cuda Clang builds pass

### PR DESCRIPTION
**Pull request by atrantan**
This patch tries to solve some issue when building hpx  with enabled Cuda Clang.

### CODE REVIEW

File: hpx/compute/cuda/default_executor.hpp [hunk 1]
```diff
@@ -188,6 +188,12 @@ namespace hpx { namespace compute { namespace cuda
             target_.synchronize();
         }
 
+        template <typename F, typename ... Ts>
+        void execute(F && f, Ts &&... ts) const
+        {
+            sync_execute(std::forward<F>(f), std::forward<Ts>(ts)...);
+        }
+
```
Comment (hkaiser) [inline hpx/compute/cuda/default_executor.hpp]: Why is this necessary?

Comment (atrantan) [inline hpx/compute/cuda/default_executor.hpp]: Needed here https://github.com/STEllAR-GROUP/hpx/blob/master/tests/performance/compute/cuda/synchronize.cpp#L43

Comment (hkaiser) [inline hpx/compute/cuda/default_executor.hpp]: Ok, let's rather change that spot to use `sync_execute()`. I must have missed changing that while re-implementing the executors.

Comment (atrantan) [inline hpx/compute/cuda/default_executor.hpp]: Done in 66c5d5858c10c327ab5757c3fb9adb64cae6190a
